### PR TITLE
Fixed bugs in Safari and IE. Closes #837 and #784

### LINKED ISSUE #837: Browser Compatibility
Run tests in IE (latest versions), and various other operating systems to check that all tests are passing.

IE11: https://gist.github.com/braydonf/61459334f5a86dc5f7be

### PR DESCRIPTION
**Pull request by braydonf**

### CODE REVIEW

**Comment by eordano** [comment]
LGTM

File: test/util/preconditions.js [hunk 1]
```diff
@@ -52,7 +52,8 @@ describe('preconditions', function() {
       $.checkArgumentType(1, PrivateKey);
     } catch (e) {
       error = e;
-      e.message.should.equal('Invalid Argument for (unknown name), expected PrivateKey but got number');
```
Comment (maraoz) [inline test/util/preconditions.js]: what is `e.message` in other browsers? I'm worried this assertion is failing there

Comment (braydonf) [inline test/util/preconditions.js]: The name of the function usually appears, in this case "PrivateKey"

Comment (maraoz) [inline test/util/preconditions.js]: "PrivateKey" instead of "(unknown name)"? in that case, ACK to this PR

Comment (eordano) [inline test/util/preconditions.js]: Yep, an extra argument should be added to the error creation

Comment (braydonf) [inline test/util/preconditions.js]: The message stays the same for browsers that support `constructor.name`, and likewise a toString() method.

**Pull request merged**